cpp/src/arrow/engine/substrait/util.cc

```diff
@@ -107,19 +111,39 @@ class SubstraitExecutor {
   bool plan_started_;
   compute::ExecContext exec_context_;
   std::shared_ptr<SubstraitSinkConsumer> sink_consumer_;
+  const ConversionOptions& conversion_options_;
 };
 
 }  // namespace
 
 Result<std::shared_ptr<RecordBatchReader>> ExecuteSerializedPlan(
-    const Buffer& substrait_buffer, const ExtensionIdRegistry* extid_registry,
-    compute::FunctionRegistry* func_registry) {
+    const Buffer& substrait_buffer, const PythonTableProvider& table_provider,
+    const ExtensionIdRegistry* registry, compute::FunctionRegistry* func_registry) {
   // TODO(ARROW-15732)
+  // retrieve input table from table provider
+  ConversionOptions conversion_options;
+  if (table_provider) {
+    NamedTableProvider named_table_provider =
+        [table_provider](
+            const std::vector<std::string>& names) -> Result<compute::Declaration> {
+      if (names.empty()) {
+        return Status::Invalid("names for NamedTable not provided");
+      }
```

Comment:
It wasn't clear to me if a name was required or not so I added https://github.com/substrait-io/substrait/pull/308

If that merges then we are fine returning an invalid status here since this would be an invalid plan.  However, we can probably reject it in `relation_internal.cc` instead so we don't have to repeat this logic for each binding.

Reply:
Yeah that make sense.